src/Otp/GoogleAuthenticator.php

```diff
@@ -72,25 +77,25 @@ public static function getKeyUri($type, $label, $secret, $counter = null, $optio
         // algorithm (currently ignored by Authenticator)
         // Defaults to SHA1
         if (array_key_exists('algorithm', $options)) {
-            $otpauth .= '&algorithm=' . $options['algorithm'];
+            $otpauth .= '&algorithm=' . rawurlencode($options['algorithm']);
         }
 
         // digits (currently ignored by Authenticator)
         // Defaults to 6
         if (array_key_exists('digits', $options)) {
-            $otpauth .= '&digits=' . $options['digits'];
+            $otpauth .= '&digits=' . rawurlencode($options['digits']);
```

Comment:
Digits can only be 6 or 8. Can you add a check for that instead of raw url encode, which doesn't do much for an int?